docs: improved going further modules documentation (#19219)
nuxt/nuxt: docs/2.guide/3.going-further/3.modules.md

})
```

### Adding Composables

If your module should provide composables, you can use the `addImports` utility to add them as auto-imports for Nuxt to resolve.

```ts
import { defineNuxtModule, addImports, createResolver } from '@nuxt/kit'

export default defineNuxtModule({
  setup(options, nuxt) {

>     const resolver = createResolver(import.meta.url);

Reviewer: ```suggestion
    const resolver = createResolver(import.meta.url)
```

    addImports({
      name: 'useComposable', // name of the composable to be used
      as: 'useComposable', 

>       from: resolver.resolve('runtime/composables/useComposable'), // path of composable 

Reviewer: ```suggestion
      from: resolver.resolve('runtime/composables/useComposable') // path of composable 
```

>     });

Reviewer: ```suggestion
    })
```

  }
})
```

Alternatively, you can add an entire directory by using `addImportsDir`.

```ts

> import { defineNuxtModule, addImportsDir,createResolver } from '@nuxt/kit'

Reviewer: ```suggestion
import { defineNuxtModule, addImportsDir, createResolver } from '@nuxt/kit'
```

export default defineNuxtModule({
  setup(options, nuxt) {

>     const resolver = createResolver(import.meta.url);
>     addImportsDir(resolver.resolve('runtime/composables'));

Reviewer: ```suggestion
    const resolver = createResolver(import.meta.url)
    addImportsDir(resolver.resolve('runtime/composables'))
```

})
```

### Adding Composables

If your module should provide composables, you can use the `addImports` utility to add them as auto-imports for Nuxt to resolve.

```ts
import { defineNuxtModule, addImports, createResolver } from '@nuxt/kit'

export default defineNuxtModule({
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    addImports({
      name: 'useComposable', // name of the composable to be used
      as: 'useComposable', 
      from: resolver.resolve('runtime/composables/useComposable'), // path of composable 
    });
  }
})

> 

Reviewer: ```suggestion
```

> export default defineNuxtModule({
>   setup(options, nuxt) {
>       const resolver = createResolver(import.meta.url);
>       addImports({
>         name: 'useComposable', // name of the composable to be used
>         as: 'useComposable', 
>         from: resolver.resolve('runtime/composables/useComposable'), // path of composable 
>       });
>   }
> })

Reviewer: ```suggestion
export default defineNuxtModule({
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    addImports({
      name: 'useComposable', // name of the composable to be used
      as: 'useComposable', 
      from: resolver.resolve('runtime/composables/useComposable'), // path of composable 
    });
  }
})
```

```

> alternative you can add an entire directory by using `addImportsDir`

Reviewer: ```suggestion

Alternatively, you can add an entire directory by using `addImportsDir`.
```

```ts

import { defineNuxtModule, addImportsDir,createResolver } from '@nuxt/kit'

> export default defineNuxtModule({
>   setup(options, nuxt) {
>       const resolver = createResolver(import.meta.url);
>       addImportsDir(resolver.resolve('runtime/composables'));
>   }
> })

Reviewer: ```suggestion
export default defineNuxtModule({
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    addImportsDir(resolver.resolve('runtime/composables'));
  }
})
```

```

> > the module starter deliberately disable auto-imports while developing a module. [https://github.com/nuxt/starter/blob/module/.nuxtrc#L1](https://github.com/nuxt/starter/blob/module/.nuxtrc#L1)

Reviewer: ```suggestion
::alert
Note that auto-imports are not enabled for files within `node_modules` for performance reasons, so the module starter [deliberately disables them](https://github.com/nuxt/starter/blob/module/.nuxtrc#L1) while developing a module. If you are using the module starter, auto-imports will not be enabled in your playground either.
```

> 
> 

Reviewer: ```suggestion

```

})
```

### Adding Composables

If your module should provide composables, you can use the `addImports` utility to add them as auto-imports for Nuxt to resolve.

```ts
import { defineNuxtModule, addImports, createResolver } from '@nuxt/kit'

export default defineNuxtModule({
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    addImports({
      name: 'useComposable', // name of the composable to be used
      as: 'useComposable', 
      from: resolver.resolve('runtime/composables/useComposable'), // path of composable 
    });
  }
})

```

Alternatively, you can add an entire directory by using `addImportsDir`.

```ts

> 

Reviewer: ```suggestion
```

import { defineNuxtModule, addImportsDir,createResolver } from '@nuxt/kit'

export default defineNuxtModule({
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    addImportsDir(resolver.resolve('runtime/composables'));
  }

> })
> 

Reviewer: ```suggestion
})
```

> 

Reviewer: ```suggestion
```

```

::alert
Note that auto-imports are not enabled for files within `node_modules` for performance reasons, so the module starter [deliberately disables them](https://github.com/nuxt/starter/blob/module/.nuxtrc#L1) while developing a module. If you are using the module starter, auto-imports will not be enabled in your playground either.

> ::

Reviewer: ```suggestion
::

```
